### ISSUE TITLE
the whole row is re-rendered on one cellData change

### ISSUE THREAD

**Issue by anytimecoder** [facebook/fixed-data-table#286]
Hi,

I've noticed that the whole row is re-rendered even if the data is changed only in one cell. This is happening because `FixedDataTableCell` component uses `PureRendererMixin` and receives `rowData` as one of the props.
Extract from the class:
```
var FixedDataTableCell = React.createClass({
  displayName: 'FixedDataTableCell',

  mixins: [ReactComponentWithPureRenderMixin], //<= this is part of the problem

  propTypes: {

    ...
    
    rowData: PropTypes.oneOfType([PropTypes.object.isRequired, PropTypes.array.isRequired]), //<=another part of the problem

```

So because every cell receives `rowData` as one of it's props - it'll always be re-rendered even if the `cellData` is the same.

One of the possible solutions would be to implement smarter `shouldComponentUpdate` method, that skips `rowData` on equality check.

**Comment by anytimecoder** [comment]
On second thought, there might be cases (I'm sure there are) where the cellRenderer (as it's external) renders cell based on RowData or ColumnData and ignoring those on shouldComponentUpdate will break the table in that particular case.
Although it still not efficient that the whole row (and probably column) is re-rendered on one cell change.

**Comment by anytimecoder** [comment]
As far as I can see this issue will be resolved in version 0.6 by providing the option of using custom components

**Comment by pieterv** [comment]
Awesome, glad to hear this. Let us know if you run into problems with this!
